src/test/java/com/xceptance/neodymium/util/SelenideAddonsTest.java

```diff
@@ -235,127 +301,115 @@ public void testWrapAssertionErrorWithoutMessage()
         }
     }
 
+    @Test()
+    public void testIsThrowableCausedBy()
+    {
+        Throwable causedByIOException = new RuntimeException("This is runtime exception", new AssertionError("this is assertion error", new IOException("this is final cause")));
+        Assert.assertTrue("Throwable has unexpected cause", SelenideAddons.isThrowableCausedBy(causedByIOException, IOException.class));
+    }
+
+    @Test()
+    public void testIsThrowableCausedByMessage()
+    {
+        Throwable causedByIOException = new RuntimeException("This is runtime exception", new AssertionError("this is assertion error", new IOException("this is final cause")));
+        Assert.assertTrue("Throwable has unexpected cause",
+                          SelenideAddons.isThrowableCausedBy(causedByIOException, IOException.class, "this is assertion error"));
+    }
+
```

Comment:
add a test method `testIsThrowableNotCausedBy` testing the negativ outcome of the function.

Reply:
Done